mediabridge/main.py

```diff
@@ -1,31 +1,49 @@
 import logging
+import os
+from datetime import datetime
 
 import typer as typer
 from tqdm.contrib.logging import logging_redirect_tqdm
 
 from mediabridge.data_processing import wiki_to_netflix
 
 
-def main(verbose: bool = typer.Option(False, "--verbose", "-v"), test: bool = False):
-    if test:
+def main(
+    verbose: bool = typer.Option(False, "--verbose", "-v"),
+    log: bool = False,
+    test: bool = False,
```

Comment:
The old behavior was that test defaulted to `True` and we passed a `--prod` flag to enable the production behavior. I'm fine with it either way.

Reply:
In wiki_to_netflix.py, the name main and prod flag checks are still there since we talked about still calling the script directly, I think the previous behavior is probably better. Maybe "--full" is a better descriptor than "prod" (since we don't know what our actual production process is going to look like)?

Also, about running script files directly, I think we should find a way so that you can still call verbose/logging flags to other scripts besides main, whether that's just being able to specify a module as an arg to the main script or maybe there's some way to specify in a script's __name__ check that this script has common arguments it should check for with main.

Reply:
I like the idea of a `--full` flag!

The logging setup, a call to `basicConfig` or other more complex configurations, propagates through the whole runtime. What that means is that any logger created with `logging.getLogger()`, from any module, will share the same logging rules. Does that answer your question?

Reply:
I'll rename it to --full then! :)

Also I meant more along the lines of what we were discussing last meeting where we should keep some other entry points besides main to run individual systems. We could solve that with some arg to main.py like --module=wiki_to_netflix to just run the wiki_to_netflix system while also handling -v or --log, but maybe there's some way to propagate cl args to submodules without duplicating logic, unless you were thinking of a different solution to running individual systems?

Reply:
Yes, we could make the "main" main a kind of wrapper/super module. It could have different commands it could run from other modules, even just calling their `main()` methods.

On the other hand, that kind of conflicts with the `pipenv run <foo>` stuff right? If we want to run module A, presumably we would do `pipenv run A`.